lib/rules/no-multi-spaces.js

```diff
@@ -7,191 +7,42 @@
-        validateBetween(node.consequent, node.alternate);
-    }
+        var comment;
 
-    /**
-     * Report if there are multiple spaces around equal operator in variable declaration.
-     * @param {ASTNode} node The node to check.
-     * @returns {void}
-     * @private
-     */
-    function checkVar(node) {
-        if (node.init) {
-            validateBetween(node.id, node.init);
-        }
-    }
+        while (lastCommentIndex < comments.length) {
 
-    /**
-     * Generates a function to check object literals and other list-like nodes.
-     * @param {String} property Name of the node's collection property.
-     * @returns {Function} A function that checks spacing within list-like node.
-     * @private
-     */
-    function checkList(property) {
-        /**
-         * Report if there are multiple spaces around list of items in objects,
-         * function parameters, sequences and declarations.
-         * @param {ASTNode} node The node to check.
-         * @returns {void}
-         * @private
-         */
-        return function(node) {
-            var items = node[property];
-
-            for (var i = 0, l = items.length; i < l - 1; i++) {
-                validateBetween(items[i], items[i + 1]);
-            }
-        };
-    }
+            comment = comments[lastCommentIndex];
 
-    /**
-     * Checks arrays literals, allowing that some elements may be empty.
-     * @param {ASTNode} node An ArrayExpression.
-     * @returns {void}
-     * @private
-     */
-    function checkArray(node) {
-        var elements = node.elements,
-            elementIndex = 0,
-            tokens = context.getTokens(node),
-            tokenIndex = 0,
-            length = tokens.length,
-            dest;
-
-        // Check spacing at the beginning, around commas, and at the end
-        while (tokenIndex < length - 1) {
-            validate(tokens[tokenIndex], tokens[++tokenIndex]);
-
-            // Don't advance to the next element until both sides of the comma
-            // have been checked
-            if (tokens[tokenIndex].value !== ",") {
-                // Skip to just before the end of the array or the next comma
-                if (elements[elementIndex]) {
-                    dest = elements[elementIndex].range[1];
-                    // Move to the end of the element
-                    while (tokens[tokenIndex].range[1] < dest) {
-                        ++tokenIndex;
-                    }
-                    // Move past anything (like closing parens) between the end
-                    // of the node and the following comma or end brace.
-                    while (
-                        tokenIndex + 1 < tokens.length &&
-                        tokens[tokenIndex + 1].value !== ","
-                    ) {
-                        ++tokenIndex;
-                    }
-                }
-                ++elementIndex;
+            if (comment.range[0] <= index && index <= comment.range[1]) {
```

Comment:
The upper bound is still inclusive here. However, since it's only off by one, and `index` is coming from `pattern.lastIndex`, and a match will have a length of at least two characters, it should still function as intended..